library/src/main/java/com/pokegoapi/api/PokemonGo.java

```diff
@@ -273,11 +294,14 @@ public AuthInfo getAuthInfo()
 	 *
 	 * @param latitude  the latitude
 	 * @param longitude the longitude
-	 * @param altitude the altitude
+	 * @param altitude  the altitude
 	 * @param accuracy  the accuracy
 	 */
 	public void setLocation(double latitude, double longitude, float altitude, double accuracy) {
 		if (latitude != this.latitude || longitude != this.longitude) {
+			if (this.latitude != Double.NaN && this.longitude != Double.NaN) {
```

Comment:
The default for each is NaN, so unless someone explicitly sets one to "Double.NaN", any time the previous check is met, this check will always be true.